src/riscv-integration.adoc

```diff
@@ -791,6 +805,24 @@ CHERI violations have the following order in priority:
 . Invalid address violation
 . Bounds violation (_Lowest_)
 
+[#mtval2,reftext="mtval2"]
+==== Machine Trap Value Register 2 (mtval2)
+
+The <<mtval2>> register is an MXLEN-bit read-write register, which is added as part of the
+Hypervisor extension. {cheri_base_ext_name} also requires the implementation of this CSR.
+
+When a CHERI fault is taken on any data memory access into into M-mode, <<mtval2>> is written
+with the MXLEN-bit address which caused the fault. This follows the existing rules for reporting
```

Comment:
The priv spec calls this register "Machine bad guest physical address.", so we probably want to clarify that we are writing the virtual address here?

Reply:
hmm in the hypervisor spec chapter it is listed as "Machine Second Trap Value (mtval2) Register" which makes a lot more sense. 

I might send a PR to the ISA manual to change that in `2.2. CSR Listing`.

Reply:
for the purpose of this spec it's the "effective address", as reported in Xtval if there's (e.g.) a page fault

Reply:
I've added the word "effective" to clarify